core/state/database.go

```diff
@@ -238,6 +238,13 @@ func (db *cachingDB) purgeLoop() {
 
 // OpenTrie opens the main account trie at a specific root hash.
 func (db *cachingDB) OpenTrie(root common.Hash) (Trie, error) {
+	//try cache first
+	if db.accountTrieCache != nil {
+		if tr, ok := db.accountTrieCache.Get(root); ok {
+			return tr.(*trie.SecureTrie), nil
```

Comment:
Shall we use `.Copy()` here instead?

The BSC implementation has the `Copy()`: https://github.com/bnb-chain/bsc/pull/257/files#diff-2002ca9f189ebd4dbe9ec777405505525bb4a67db44536243e615f66d85426d2R191

Reply:
exactly, updated, thx

Reply:
exactly, updated, thx